core-common/src/main/java/org/mapstruct/IterableMapping.java

```diff
@@ -52,4 +55,11 @@
      * @return the qualifiers
      */
     Class<? extends Annotation>[] qualifiedBy() default { };
+
+    /**
+     * Specifies the result type of the element mapping method to be used in case multiple mapping methods qualify.
+     *
+     * @return the resultType to select
+     */
+    Class<?> resultType() default void.class;
```

Comment:
Ok. I'll adapt. But now we have an inconsistency for `@BeanMapping#resultType` and `@Mapping#resultType` which do in essence the same.  That's also confusing for the user, wouldn't you agree?